sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -190,12 +190,16 @@ def __init__(
         self._credential = credential
         if not name:
             raise ValueError("name can not be None or empty")
-        if not isinstance(name, str):
-            raise TypeError("name must be a string.")
+        if not isinstance(name, (str, Tuple)):
+            raise TypeError("name must be a string or a tuple of strings.")
         self._name = name
 
     def on_request(self, request):
-        request.http_request.headers[self._name] = self._credential.key
+        if isinstance(self._name, str):
+            request.http_request.headers[self._name] = self._credential.key
+        else:
+            for name, key in zip(self._name, self._credential.key):
```

Comment:
What does the error look like if the tuple of `name` does not match the tuple of `keys`?

Reply:
In my opinion, we want SDK developers to do the check, e.g. the ctor of MetricsAdvisorKeyCredential because they have better ideas.

Reply:
(Assuming you are asking the case the len(name) != len(key)) it will use the smaller one.

e.g.
_name = ('header1', 'header2', 'header3')
_key = ('key1', 'key2')

it only updates 'header1' with 'key1', 'header2' with 'key2' and ignore 'header3'